### PR TITLE
Improve scheduler CLI description

### PR DESCRIPTION
**Pull request by dharmab**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design

/kind documentation

> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

This PR adds a few explanatory sentences to the kube-scheduler CLI description (which is in turn used to generate the documentation website). To understand why this is needed, let's look at the first few lines of the documentation for other core CLI components from the perspective of a new user or contributor:

> [The kubelet is the primary “node agent” that runs on each node.](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/)

> [The Kubernetes network proxy runs on each node. This reflects services as defined in the Kubernetes API on each node and can do simple TCP, UDP, and SCTP stream forwarding or round robin TCP, UDP, and SCTP forwarding across a set of backends.](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/)

> [The Kubernetes controller manager is a daemon that embeds the core control loops shipped with Kubernetes.](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/)

> [The Kubernetes API server validates and configures data for the api objects which include pods, services, replicationcontrollers, and others. The API Server services REST operations and provides the frontend to the cluster’s shared state through which all other components interact.](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/)

So far, not bad. Some of the descriptions are a little wordy, but in general from reading the first one or two sentences, a new user reading the docs has a basic idea of what each component does.

What about kube-scheduler?

> The Kubernetes scheduler is a policy-rich, topology-aware, workload-specific function that significantly impacts availability, performance, and capacity. The scheduler needs to take into account individual and collective resource requirements, quality of service requirements, hardware/software/policy constraints, affinity and anti-affinity specifications, data locality, inter-workload interference, deadlines, and so on. Workload-specific requirements will be exposed through the API as necessary.

This... is very confusing. A new user reading this doesn't learn what the scheduler actually does, just that it seems to do... _something important?_

I've copied a couple of sentences from https://kubernetes.io/docs/concepts/scheduling/kube-scheduler/#scheduling to clarify the documentation here.

I wonder if the entire paragraph quoted above is really entirely necessary? It reads like the opening paragraph of a requirements specification doc, not the quick help text most people expect from a CLI tool.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kube-scheduler/app/server.go
Comment (sftim) [inline cmd/kube-scheduler/app/server.go]: Things I'd want readers to learn, instead of this:
- you can have more than one scheduler in your cluster (for high availability)
- you can have more than one scheduler in your cluster (if you deploy multiple, different schedulers, and configure things)
- kube-scheduler is part of the control plane
- kube-scheduler works by first filtering out nodes where the Pod can't possibly be placed, and then scoring what's left to find a good (good enough?) fit
   - https://kubernetes.io/docs/concepts/configuration/scheduling-framework/ talks about the full cycle, but I'd not mention that
- kube-scheduler considers multiple constraints when scheduling, but I wouldn't list which ones (at least not here)

Comment (dharmab) [reply cmd/kube-scheduler/app/server.go]: I've updated this with a more substantial rewrite to incorporate these points while still being short enough to understand.

**Review by sftim — COMMENTED** [review]
/lgtm

**Comment by dharmab** [comment]
@sftim Thanks for the reviews, anything else needed to get this merged?

**Comment by Huang-Wei** [comment]
@dharmab  LGTM. Could you squash the commits into one? and afterwards we will get it merged.

**Comment by zacharysarah** [comment]
/lgtm

@dims @cblecker 👋 What's the path to merge here?

**Comment by cblecker** [comment]
@zacharysarah The author has been asked to squash the two commits in the PR into a single commit by an approver: https://github.com/kubernetes/kubernetes/pull/88371#issuecomment-606265020

I assume when this is done, @Huang-Wei will approve the PR as an approver for this part of the code base.

**Comment by dharmab** [comment]
Thanks for looking at this @Huang-Wei and @zacharysarah . I've squashed the commits.

**Review by damemi — APPROVED** [review]
/lgtm

**Comment by Huang-Wei** [comment]
@dharmab could you take a look at the CI failures?

**Comment by Huang-Wei** [comment]
/hold
until CI failures get resolved.

**Comment by dharmab** [comment]
Rebasing to see if that resolves the CI failure

**Comment by dharmab** [comment]
> [measurement call PodStartupLatency - PodStartupLatency error: pod startup: too high latency 99th percentile: got 7.323477279s expected: 5s]

This looks like an unrelated failure. Can that test be ignored?

**Pull request merged**
